include/FanFicArchive.py

```diff
@@ -0,0 +1,85 @@
+from bs4 import BeautifulSoup; from urllib import request; from re import match; import requests
+
+parser = "lxml"
+
+def fanfic_req(link): return BeautifulSoup(request.urlopen(link).read(), parser)
+def fanfic_check(id, name):
+    if isinstance(id, int): id = str(id)
+    else: assert isinstance(id, str), "Error for" + name + ": id not int or str"
+
+def ao3_rip(ao3_id): # downloads whole story in one go
+    fanfic_check(ao3_id, "ao3_rip")
+    text = fanfic_req('http://archiveofourown.org/works/'+ao3_id+'?view_adult=true&view_full_work=true')
+    tag_bar = text.find("dl", class_="stats").find_all("dd") # everything in the top bar
+    tag_list = {}
```

Comment:
`list` in the name but type is a dict

Reply:
practical use rather than the data type, as it is going to be used as a key-value set.